app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java

```diff
@@ -18,23 +20,24 @@
       R.layout.activity_request_state_declined
   };
 
-  private SelectShopperActivity.SelectShopperAdapter.ShopRequest request;
+  private OutgoingRequest request;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     request = getIntent().getParcelableExtra(EXTRAS_REQUEST_KEY);
-    setContentView(LAYOUTS[request.status.ordinal()]);
+    setContentView(LAYOUTS[request.getStatus().ordinal()]);
     ButterKnife.inject(this);
   }
 
   @Optional
   @OnClick(R.id.request_state_cancel_button)
   public void onCancelRequest(View view) {
+    request.setStatus(BaseRequest.RequestStatus.CANCELED);
     // TODO: Cancel request in server
-    request.shopper = null;
+    OutgoingRequest newRequest = new OutgoingRequest(request.getShoppingList());
     Intent intent = new Intent(this, SelectShopperActivity.class);
-    intent.putExtra(SelectShopperActivity.EXTRAS_REQUEST_KEY, request);
+    intent.putExtra(SelectShopperActivity.EXTRAS_REQUEST_KEY, newRequest);
```

Comment:
That's one of way doing it.
I'm still not sure about the flow though. Technically, `SelectShopperActivity` does something very specific, and you can use it with `startActivityForResult`, but the same also applies to `CreateShoppingListActivity`.
So ideally, both of them would receive the information they need (List<Contact> and ShoppingList respectively) and return what the user did (specific contact, updated shopping list object) and you would have some manager activity that takes care of the flow. That would make more sense to me because there isn't really a predefined order to these 2 activities (and after a request is cancelled, you don't restart the entire process, the flow is 1->2->3->1->3).
However, I'm not sure about that. It could be that we need to move the shopping list to be the first activity, and then the flow would be 1->2->3->2->3.

In any case, I'd like to limit the usage of complex objects where we don't need them. That always leads to trouble.